src/Datagrid/ProxyQuery.php

```diff
@@ -202,6 +202,7 @@ public function execute(array $params = [], $hydrationMode = null)
 
         $query = $this->getDoctrineQuery();
 
+        // NEXT_MAJOR: Remove this block.
         foreach ($this->hints as $name => $value) {
             $query->setHint($name, $value);
         }
```

Comment:
We could move maybe this`foreach` at the end of `getDoctrineQuery` in `4.x` so we can remove it from `SmartPaginatorFactory` (not sure if it is used though).

Reply:
I'm wonder if it's not better to deprecate it since it's not part of the API...

Reply:
It was added in https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/730 which seems legit.

Reply:
Ok